lang/en.json

```diff
@@ -850,6 +903,63 @@
     "Half": "Half"
   }
 },
+"DND5E.DAMAGE": {
+  "FIELDS": {
+    "number": {
+      "label": "Die Number",
+      "hint": "Number of dice to roll."
+    },
+    "denomination": {
+      "label": "Die Denomination",
+      "hint": "Denomination of the dice to roll."
+    },
+    "bonus": {
+      "label": "Damage Bonus",
+      "hint": "Bonus added to the damage roll."
+    },
+    "types": {
+      "label": "Damage Types",
+      "hint": "Type of damage inflicted or multiple for the user to select from."
+    },
+    "custom": {
+      "label": "Custom Damage Formula",
+      "enabled": {
+        "label": "Enable Custom Formula",
+        "hint": "Should the custom formula be used rather than the default dice."
+      },
+      "formula": {
+        "label": "Damage Formula",
+        "hint": "Custom damage formula."
+      }
+    },
+    "scaling": {
+      "label": "Damage Scaling",
+      "mode": {
+        "label": "Scaling Mode",
+        "hint": "Method by which the scaling increase is calculated."
+      },
+      "number": {
+        "label": "Dice Scaling",
+        "hint": "Number of dice to increase for each scaling step. Will be applied to the first die found in the damage formula if more than one is present."
+      },
+      "formula": {
+        "label": "Scaling Formula",
+        "hint": "Arbitrary scaling formula that will be multiplied for each scaling step and added to the original formula."
+      }
+    }
+  },
+  "Part": {
+    "Action": {
+      "Create": "Create Damage Part",
+      "Delete": "Delete Damage Part"
+    }
+  },
+  "Scaling": {
+    "Half": "Every Other Level",
+    "None": "No Scaling",
+    "Whole": "Every Level"
+  }
```

Comment:
Wondering if we shouldn't simply bite the bullet and allow for manual input of damage at every level. [This issue](<https://github.com/foundryvtt/dnd5e/issues/313#issuecomment-1154098007>) had some good suggestions, as well as edge cases that half/none/every level doesn't cover.

Reply:
Some of the more awkward spells where individual damage parts scale, like *flame strike* or *ice storm*, should be handled now that scaling is attached to the damage part in the model. I don't much relish the idea of building a UI for per-level scaling. Isn't it already achievable with only the scaling formula and whatever the equivalent of `@item.level` will be?

Reply:
Yeah, per-type scaling like this handle basically 99% of official spells, and a could of edge cases can be handled by setting the formula explicitly and using the scaling value in roll data. There are a few edge cases outside of damage (like durations that change from `1 minute` => `10 minutes` => `1 hour` as you upcast), but none of those are super urgent.

Reply:
Yeah there aren't many left. I am mainly thinking of spells like Shadow Blade that don't scale in any easy pattern. For example that spell stops scaling at 7th level.

Reply:
`BaseActivityData` has a `consumption.scaling.max` which should handle those maximums now. If the per-level scaling is just for UX purposes, and not because we can't accommodate that scaling with a formula then yeah, we can consider it. It's a non-trivial amount of UI work though, so unlikely to make it into 4.0.